piff/polynomial_interp.py

```diff
@@ -0,0 +1,389 @@
+    def _generate_indices(self, order):
+        """Generate, for internal use, the exponents i,j used in the polynomial model
+        p(u,v) = sum c_{ij} u^i v^j
+
+        This needs to be called whenever the order of the polynomial fit is 
+        changed. At the moment that is just when an object is initialized or
+        updated from file.
+
+        :param order:   The maximum order of the polynomial; the max value of
+                        i+j where p(x,y) ~ x^i y^j
+         """
+        indices = []
+        for p in xrange(order+1):
+            for i in xrange(p+1):
+                j = p-i
+                indices.append((i,j))
+        return indices
+
+    def _pack_coefficients(self, parameter_index, C):
+        """Pack the 2D matrix of coefficients used as the model fit parameters
+        into a vector of coefficients, either so this can be passed as a starting
+        point into the curve_fit routine or for serialization to file.
+
+        For subclasses, the 2D matrix format could be whatever you wanted as long
+        as _initialGuess, _interpolationModel, and the pack and unpack functions are 
+        consistent. The intialGuess method can return and the _interpolationModel can
+        accept parameters in whatever form you like (e.g. could be a dict if you want)
+        as long as _pack_coefficients can convert this into a 1D array and _unpack_coefficients
+        convert it the other way.
+
+        :param parameter_index: The integer index of the parameter; the lets us 
+                                find the order of the parameter from self.
+        :param C:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+        :returns coeffs:    A 1D numpy array of coefficients of length self.nvariable
+        """
+        coeffs = numpy.zeros(self.nvariables[parameter_index])
+        for k,(i,j) in enumerate(self.indices[parameter_index]):
+            coeffs[k] = C[i,j]
+        return coeffs
+
+
+    def _unpack_coefficients(self, parameter_index, coeffs):
+        """Unpack a sequence of parameters into the 2D matrix for the 
+        given parameter_index (which determines the order of the matrix)
+
+        This function is the inverse of _pack_coefficients
+                           
+        :param parameter_index: The integer index of the parameter being used
+        :param coeffs:     A 1D numpy array of coefficients  of length self.nvariable
+        :returns:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+
+        """
+        k=0
+        n=self.orders[parameter_index]+1
+        C = numpy.zeros((n, n))
+        for k,(i,j) in enumerate(self.indices[parameter_index]):
+                C[i,j] = coeffs[k]
```

Comment:
Minor comment here, but this seems to have gotten extra indent spaces for some reason.